apiclient/apiclient.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2019 Thomas Weber
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiclient
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+)
+
+const (
+	PODMAN_API_VERSION = "v1.0.0"
+)
+
+type APIClient struct {
+	baseUrl    string
+	httpClient *http.Client
+	logger     hclog.Logger
+}
+
+func NewClient(logger hclog.Logger) *APIClient {
```

Comment:
`GuessSocketPath` sounds like we are guessing :) Thoughts on passing in a an Config struct, that contains a logger and socket path? when we initialize the config we can use a renamed version of GuessSocketPath as DefaultSocketPath is one isn't provided

Reply:
Method is now renamed to DefaultSocketPath().

Not sure how to approach the config struct. What about moving api initialization from NewPodmanDriver() to SetConfig()? This way we could simply share the drivers config with api or derive a separate struct from driverconfig.

Reply:
IMO something like this would be nice, if `SetConfig` is the entry point it could be built up there, We can provide a `DefaultConfig()` method in the api package if we are initializing the client first, then calling setconfig but that seems like something to think about for future improvements to only call once

```go
type ClientConfig struct{
         SocketPath string
}

func NewClient(cfg ClientConfig, logger hclog.Logger) *APIClient {
```